akka-http-core/src/main/scala/akka/http/impl/settings/HostConnectionPoolSetup.scala

```diff
@@ -4,6 +4,9 @@
 
 package akka.http.impl.settings
 
+import akka.annotation.InternalApi
+
 /** INTERNAL API */
+@InternalApi
 final case class HostConnectionPoolSetup(host: String, port: Int, setup: ConnectionPoolSetup)
```

Comment:
Currently, this class is returned by `akka.http.javadsl.HostConnectionPool#setup`
And then, used in some private methods and a `private[http] final clase class` under `akka.http.scaladsl.Http`

So I guess, it's not really an InternalApi